### ISSUE TITLE
overhang cuts of the top

### ISSUE THREAD

**Issue by sum-catnip** [Droggelbecher/bevy-fast-tilemap#20]
Hi!

i was trying to use overhang (perspective) to have tiles visually appear bigger.
for reference, we have those tiles:
![tiles](https://github.com/Droggelbecher/bevy-fast-tilemap/assets/26556802/bf84383e-5c72-4b70-814a-c7f3e983e906)

to see it easier here is the same thing with a black grid:
![Tiles_for_export_and_grid](https://github.com/Droggelbecher/bevy-fast-tilemap/assets/26556802/395f8aa1-b322-4773-95a4-01f960af2775)

each tile is 40x20, i used an overhang of
```rust
            vec2(4. * TILE_SIZE.x, 4. * TILE_SIZE.y), // inner
            vec2(2. * TILE_SIZE.x, 4. * TILE_SIZE.y), // lefttop
            vec2(2. * TILE_SIZE.x, 0. * TILE_SIZE.y), // rightbottom
```

so basically
- 4 tiles padding between each tile
- 2 tiles padding from the left and right
- 4 tiles padding from the top
- 0 tiles padding from the bottom

this is what im seeing:
![image](https://github.com/Droggelbecher/bevy-fast-tilemap/assets/26556802/4fb0ec09-a582-4850-ab0e-94d68f2a8dc0)

only 4 tiles (2 horizontal, 2 vertical) are shown of the palm.

**Comment by sum-catnip** [comment]
i tried fixing it in the shader code by not doing:
> var max_overhang = map.inner_padding / 2.0;

but that didn't really work. i dont understand enough about the shader to fix it

**Comment by Droggelbecher** [comment]
Nice tiles :)
The root cause is that the shader code looks only in directly neighbouring tiles for overhang, i.e. it assumes that the overhang is not bigger than one tile (straight or diagonal). We could change that, but then the shader would need to check a lot of neighbouring tiles on whether they contribute to the current one due to overhangs.

You could consider a second map layer that represents another height level (i.e. cut the tree into a lower and upper part); but depending on what you're building that may introduce some unwanted complexity to your code.

Maybe there is a solution that is both performant and easy to use, let me think about it a bit ...

**Comment by sum-catnip** [comment]
haha thank you :D
and thanks so much for the well thought out response, that makes perfect sense!

i think the best approach for us would be
> You could consider a second map layer that represents another height level (i.e. cut the tree into a lower and upper part); but depending on what you're building that may introduce some unwanted complexity to your code.

so im gonna go with that that.

**Comment by sum-catnip** [comment]
![image](https://github.com/Droggelbecher/bevy-fast-tilemap/assets/26556802/bb86c23a-e25b-463b-91a2-6127609e725f)
yay :)
